dev-scheme/chicken/chicken-4.13.0.recipe

```diff
@@ -0,0 +1,120 @@
+SUMMARY="A Scheme interpreter and native Scheme to C compiler"
+DESCRIPTION="CHICKEN is a compiler for the Scheme programming language. \
+CHICKEN produces portable and efficient C, supports almost all of the R5RS \
+Scheme language standard, and includes many enhancements and extensions. \
+CHICKEN runs on Linux, MacOS X, Windows, and many Unix flavours."
+HOMEPAGE="http://www.call-cc.org/"
+REVISION="1"
+LICENSE="BSD (2-clause)"
+COPYRIGHT="2008-2017 The Chicken Team
+		2000-2007 Felix L. Winkelmann"
+SOURCE_URI="http://code.call-cc.org/releases/$portVersion/chicken-$portVersion.tar.gz"
+CHECKSUM_SHA256="add549619a31363d6608b39e0cf0e68b9d5e6ff2a719b5691ddeba57229c6c43"
+
+GLOBAL_WRITABLE_FILES="
+	non-packaged/lib/chicken/8 directory keep-old
+	"
+
+ARCHITECTURES="!x86_gcc2 x86 ?x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+# On x86_gcc2 we don't want to install the commands in bin/<arch>/, but in bin/.
+commandSuffix=$secondaryArchSuffix
+commandBinDir=$binDir
+if [ "$targetArchitecture" = x86_gcc2 ]; then
+	commandSuffix=
+	commandBinDir=$prefix/bin
+fi
+
+PROVIDES="
+	chicken$secondaryArchSuffix = $portVersion
+	cmd:csi$commandSuffix = $portVersion
+	cmd:csc$commandSuffix = $portVersion
+	cmd:chicken$commandSuffix = $portVersion
+	cmd:chicken_bug$commandSuffix
+	cmd:chicken_install$commandSuffix
+	cmd:chicken_profile$commandSuffix = $portVersion
+	cmd:chicken_status$commandSuffix
+	cmd:chicken_uninstall$commandSuffix
+	cmd:feathers$commandSuffix
+	lib:libchicken$secondaryArchSuffix = 8 compat >= 8
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	"
+
+PROVIDES_devel="
+	chicken${secondaryArchSuffix}_devel = $portVersion
+	devel:libchicken$secondaryArchSuffix = 8 compat >= 8
+	"
+REQUIRES_devel="
+	chicken$secondaryArchSuffix == $portVersion base
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	"
+BUILD_PREREQUIRES="
+	cmd:gcc$secondaryArchSuffix
+	cmd:ld$secondaryArchSuffix
+	cmd:make
+	"
+
+PATCH()
+{
+if [ "$targetArchitecture" = x86_gcc2 ]; then
+	echo "Removing -fwrapv flag for gcc2 build"
+	sed -i 's/-fwrapv//' Makefile.haiku
+fi
+	echo "Removing git check"
+	sed -i 's/sh $(SRCDIR)identify.sh/echo/' rules.make
+}
+
+TEST_REQUIRES="
+	cmd:diff
+	"
+
+BUILD()
+{
+	# odd timestamp issue
+	touch build-version.c
+
+	make PLATFORM=haiku PREFIX=$prefix \
+	BINDIR=$commandBinDir \
+	LIBDIR=$libDir \
+	SHAREDIR=$dataDir \
+	MANDIR=$manDir \
+	DOCDIR=$docDir \
+	INCLUDEDIR=$includeDir \
+	EGGDIR=$prefix/non-packaged/lib/chicken/8 \
+	HOSTNAME=shredder $jobArgs
```

Comment:
I'd use something like:
```bash
	make $jobArgs PLATFORM=haiku \
		PREFIX=$prefix \
		BINDIR=$commandBinDir \
		LIBDIR=$libDir \
		SHAREDIR=$dataDir \
		MANDIR=$manDir \
		DOCDIR=$docDir \
		INCLUDEDIR=$includeDir \
		EGGDIR=$prefix/non-packaged/lib/chicken/8 \
		HOSTNAME=shredder
```

Reply:
No $jobArgs?

Reply:
I did not drop it; I moved it to the first line :)
(But this is just a suggestion.)
BTW, for `HOSTNAME`, in case it does not require a value, maybe we should avoid a hardcoded (and arbitrary?) name.

Reply:
Ah, missed that one :) for HOSTNAME it was in the recipe, not sure if it's really needed ...